unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,27 @@
+#lang racket/base
+
+(provide hash-pluck-lens)
+
+(require racket/list
+         lens/base/main
+         "hash.rkt"
+         "compound.rkt")
+(module+ test
+  (require rackunit))
+
+(define (hash-pluck-lens . ks)
+  (apply compound-hash-lens
+         (append-map
+          (λ (k)
+            (list k (hash-ref-lens k)))
+          ks)))
+
+(module+ test
+  (check-equal? (lens-view (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4 'c 5))
+                (hash 'a 4 'b 2 'c 5))
+  (test-case "this one currently fails"
+    (check-equal? (lens-set (hash-pluck-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'a 4))
+                  (hash 'a 4 'b 2 'c 3)))
```

Comment:
That would for the struct, I'm wondering if it would need tweaking for the generic lenses. I think that reveals that the lens is a struct in error messages. Is there a way to make it more opaque?